src/main/java/com/parasoft/findings/sonar/soatest/SurefireSOAtestParser.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.parasoft.findings.sonar.soatest;
+
+import com.parasoft.findings.sonar.Logger;
+import com.parasoft.findings.sonar.Messages;
+import com.parasoft.findings.sonar.exception.InvalidReportException;
+import com.parasoft.findings.sonar.soatest.data.UnitTestClassReport;
+import com.parasoft.findings.sonar.soatest.data.UnitTestIndex;
+import com.parasoft.findings.utils.common.nls.NLS;
+import org.apache.commons.lang3.StringUtils;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.measures.Metric;
+import org.sonar.api.scanner.ScannerSide;
+
+import javax.annotation.CheckForNull;
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+// Adapted from https://github.com/SonarSource/sonar-java/blob/c6c9fc74e75ab3c25f46d3d45cf4f1d9de80bb27/java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java
+@ScannerSide
+public class SurefireSOAtestParser {
```

Comment:
I know we copied the name pattern from original code or sonar-java. But I think it's better to call this `XUnitSOAtestParser` and call all `UnitXXX` classes as `XUnitXXX` in our case.